### DISCUSSION TITLE
Deploy next app with custom express server

### DISCUSSION THREAD

**Discussion by goncy** [vercel/vercel#4588] [Help] [answered]
I have a typescript express server that runs next, when i run `now dev` it works fine, but when deploying, only the next files get deployed.

![image](https://user-images.githubusercontent.com/6494462/83945758-306bb000-a7e3-11ea-837c-d61a2cb8c8f3.png)

this is my settings configuration

![image](https://user-images.githubusercontent.com/6494462/83945797-73c61e80-a7e3-11ea-9e3d-86b35e6ea38c.png)

and this is my dist folder

But when deploying this is the output

![image](https://user-images.githubusercontent.com/6494462/83945811-85a7c180-a7e3-11ea-802a-8522f3a1cba7.png)

**Comment by steveklabnik** [comment]
I am also seeing this :/

**Comment by marcperrinp** [comment]
I asked a similar question and got this response:

![image](https://user-images.githubusercontent.com/48444523/84240952-1097fd00-aaff-11ea-9884-c0cd7acfafb8.png)

What I don't understand is: does it mean [this part of Nextjs](https://nextjs.org/docs/advanced-features/custom-server) is unusable with Vercel?

**Comment by adriandmitroca** [comment]
Just a bit off topic but I am super curious, what made you to migrate to custom Express server that couldn't be achieved by built-in serverless functions support?

**Reply by marcperrinp** [reply]
Easy maintenance mode. (By maintenance mode, I mean blocking every page of my app by a static "maintenance screen".)

If you have any idea how to achieve that properly, I'd love to hear it! :)

**Reply by adriandmitroca** [reply]
Roughly it sounds like a doable thing, however this would make every page to be SSR (which in some cases may lead to extra complexity / Vercel limits as well).

Something like maintenance mode check (by requesting your API) in the main Layout component and then render maintenance screen depending on the response, how does that sound to you?

**Reply by marcperrinp** [reply]
I guess that could do the trick. But it felt like there had to be a simpler way to do this..

Heroku has a simple way to [enable maitenance mode](https://devcenter.heroku.com/articles/maintenance-mode).

I'm new to Vercel, so I may not be thinking the right way.

Right now I'm doing something else which isn't great, but it's easy of use.

Thanks for your help though ! If you have any idea about my [other question](https://github.com/vercel/vercel/discussions/4614), this is where I really need help! D:

**Reply by marcperrinp** [reply]
For anyone interested in maintenance mode, just wanted to highlight a [great answer on SO](https://stackoverflow.com/questions/62264995/how-do-i-put-a-nextjs-app-in-maintenance-mode-using-vercel/62310021#62310021) from @styfle

I still need to understand where to put my `maintenance.html` file, but I'm sure I'll get there!

**Reply by HunderlineK** [reply]
There's also any sort of websocket requirements (though AWS gateway does support it; so maybe it'll land in vercel at some point too)

**Comment by mcsdevv** [comment, marked as answer]
Hey all, the Vercel platform does not support using the custom Next.js server.

We recommend using [API Routes](https://nextjs.org/docs/api-routes/introduction) as the best way to make use of server-side behavior.

**Reply by goncy** [reply]
Great, thanks! I found a workaround after talking with Belen from the Vercel team, I wrote this post (it's in spanish) but it may help someone https://blog.goncy.now.sh/vercel-node-express

**Reply by condinoaljoseph** [reply]
Hi! I want to read your blog post do you have an English version of this?

**Reply by goncy** [reply]
Not really, but I think google translate should do the trick
